pkg/broker/util.go

```diff
@@ -43,6 +43,82 @@ func toBrokerPlans(apbPlans []apb.Plan) []Plan {
 	return brokerPlans
 }
 
+func updateMetadata(metadata map[string]interface{}, params []apb.ParameterDescriptor) map[string]interface{} {
+	if params == nil || len(params) == 0 {
+		return metadata
+	}
+
+	if metadata == nil {
+		metadata = make(map[string]interface{})
+	}
+
+	formDefinition := make([]interface{}, 0)
+
+	for paramIdx := 0; paramIdx < len(params); {
```

Comment:
As an alternative to returning a new paramIdx, those functions could return the number of items consumed and the loop could add it to the paramIdx
```go
if pd.DisplayGroup == "" {
  formItem, numItems = createUIFormItem(pd, paramIdx)
} else {
  formItem, numItems = createUIFormGroup(params, pd.DisplayGroup, paramIdx)
}
paramIdx = paramIdx + numItems
```